odc/geo/cog/_tifffile.py

```diff
@@ -776,14 +816,21 @@ def save_cog_with_dask(
             **upload_params,
         )
 
-    # Upload tiles
-    tiles_write_order = _tiles[::-1]  # Reverse tiles for writing
-    if len(tiles_write_order) > 4:  # Optimize for larger datasets
+    # Reverse tile order for writing: highest overview first
+    tiles_write_order = _tiles[::-1]
+
+    # Optionally merge the top N compressed overviews
+    # to reduce overhead when many overview levels exist.
```

Comment:
* for S3 output, there is a limit of 10,000 upload chunks per object stored
* Each overview level is an independent *"sub-stream"* (or maybe several, need to check code)
* Each *"sub-stream"* needs to be allocated a subset of the available chunk space ahead of time

This "optimization" is more about reducing number of these "upload sub-streams", so that remaining sub-streams have more breathing room.